pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,33 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Machine Learning <framework name>:<git tag>
+	// The values could be: "tensorflow:v2.2.0","pytorch:v1.5.1","sklearn:0.23.1","onnx:v1.7.0","xgboost:v1.1.1", "myawesomeinternalframework:1.1.0" etc.
+	Framework string `json:"framework"`
+	// Machine Learning model type, this field is used to match explainer type
+	// +kubebuilder:webhooks:Enum={"tabular","text","image"}
+	Type string `json:"type,omitempty"`
+	//Defaults to requests and limits of 1CPU, 2Gb MEM.
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Should we even default to 1CPU requested? As I understand it, memory requests make sense here and not much else. Should we simply make this memory rather than full resource requirements?

Reply:
I discussed with @yuzisun whether limits are safe as Tensorflow will by default try to use all cores and you might get CPU throttling if you limit to 1CPU. Maybe we should at least remove limits and users can add these as needed? Not sure whether we also want to set:
```
--tensorflow_intra_op_parallelism= 
--tensorflow_inter_op_parallelism=
```
But this is a separate issue.

Reply:
> Should we even default to 1CPU requested? As I understand it, memory requests make sense here and not much else. Should we simply make this memory rather than full resource requirements?

I think you are right, I am changing it to only has memory now.

Reply:
> I discussed with @yuzisun whether limits are safe as Tensorflow will by default try to use all cores and you might get CPU throttling if you limit to 1CPU. Maybe we should at least remove limits and users can add these as needed? Not sure whether we also want to set:
> 
> ```
> --tensorflow_intra_op_parallelism= 
> --tensorflow_inter_op_parallelism=
> ```
> 
> But this is a separate issue.

Clive, this resource field is meant to be the amount of resource will be needed by a model. I think users can only estimate memory usage when they create the TrainedModel CR. I will change it to only has memory.